operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/DefaultEventSourceManager.java

```diff
@@ -43,12 +45,14 @@
     }
   }
 
-  @SuppressWarnings({"rawtypes", "unchecked"})
-  public <R extends CustomResource<?, ?>> DefaultEventSourceManager(
-      ResourceController controller, ControllerConfiguration configuration, MixedOperation client) {
-    this(new DefaultEventHandler(controller, configuration, client), true);
-    registerEventSource(
-        CUSTOM_RESOURCE_EVENT_SOURCE_NAME, new CustomResourceEventSource<>(client, configuration));
+  public DefaultEventSourceManager(ConfiguredController<R> controller) {
+    this(new DefaultEventHandler<>(controller), true);
```

Comment:
This is just a remark for the future, maybe to discuss separately. It's little bit messy how we wire up and/or instantiate these object DefaultEventHandler here and the DefaultEventSourceManager in the Configured controller. Those should be probable more explicit somewhere, making the code more readable, since these two classes are quite fundamental. Now it's quite hard to trace back, what is instantiated where.

Reply:
Yes, this still needs to be cleaned up but this PR is already doing too much as it stands so that work should be done in subsequent PRs.